includes/alert.js

```diff
@@ -0,0 +1,410 @@
+/**
+ * Buoy Alert
+ *
+ * @file All of Buoy's client-side alert-handling JavaScript.
+ *
+ * @copyright Copyright (c) 2015-2016 by Meitar "maymay" Moscovitz
+ *
+ * @license GPL-3.0
+ */
+
+// We use an IIFE here to protect the page's global scope.
+(function () { // begin immediately-invoked function expression (IIFE)
+    // The Buoy user interface is primarily plain, old-school HTML5.
+    // There's nothing fancy about it, and much of it will still work
+    // correctly without JavaScript enabled. However, to offer a slicker
+    // visual user experience, we use BackboneJS to progressively enhance
+    // the static HTML content.
+    //
+    // Backbone helps us structure our interface code into "Views" that
+    // can be composed together. This means one View can contain another
+    // View inside it. Each interface element that we display in the HTML
+    // (like buttons, forms, and so on) are associated with a View at
+    // runtime by code in this file. Interactions that require JavaScript
+    // to work, like showing and hiding dialogues or other visual effects
+    // of a similar nature, are performed by calling methods on the View
+    // objects defined below.
+
+    // --------------------- //
+    // VIEWS (UI components) //
+    // --------------------- //
+
+    // Gotta declare 'em all.
+    var ActivateAlertForm,
+        ChooseTeamsPanel,
+        ContextualAlertButton,
+        ContextualAlertModal,
+        ImmediateAlertButton,
+        SubmittingAlertModal,
+        TimedAlertButton,
+        TimedAlertModal;
+
+    /**
+     * Activate Alert Form
+     *
+     * This is the containing element for the "Activate Alert" screen.
+     *
+     * @see {@link https://github.com/betterangels/buoy/wiki/Alerts#accessing-the-activate-alert-screen}
+     */
+    BUOY.Views.ActivateAlertForm = Backbone.View.extend({
+        'events': {
+            'submit': 'handleSubmit'
+        }
+    });
+
+    /**
+     * Immediate Alert Button
+     *
+     * @see {@link https://github.com/betterangels/buoy/wiki/Alerts#immediate-alerts}
+     */
+    BUOY.Views.ImmediateAlertButton = Backbone.View.extend({
+        'events': {
+            'click': function () {
+                this.$el.prop('disabled', true); // prevent double-clicks
```

Comment:
I didn't notice this before but shit thank you for preventing double submissions. So many annoying bugs come from that.

Since we do this a lot, I spent a while considering whether a simple helper function could be extracted to do this, but ultimately decided I don't think it's worth the added complexity.